clvm-traits/src/lib.rs

```diff
@@ -27,37 +27,157 @@ pub use to_clvm::*;
 pub use wrappers::*;
 
 #[cfg(test)]
-#[cfg(feature = "derive")]
-mod tests {
+pub mod tests {
     extern crate self as clvm_traits;
 
-    use std::fmt;
+    use super::*;
+
+    #[derive(Clone)]
+    pub enum TestNode {
+        Atom(usize),
```

Comment:
This won't work for tests that have atoms that exceed this, such as `i128::MAX`. Maybe either `BigInt` or `Vec<u8>` is more appropriate here?

Reply:
my first attempt (trying to keep this as simple as possible) was to store the atom data in the `Node` itself. However, this is not possible with our current interfaces/traits. When we ask an allocator for the bytes, it has to return a slice whose lifetime is tied to the allocator. You can't make it return a slice whose lifetime is tied to the node itself (because that violates the trait).

So, this `usize` is an index into the `atoms` member of the `TestAllocator`

Reply:
Ok, we can probably improve the test API later once this is all sorted out